core/trino-main/src/main/java/io/trino/operator/scalar/DateTimeFunctions.java

```diff
@@ -264,6 +264,36 @@ public static long truncateDate(@SqlType("varchar(x)") Slice unit, @SqlType(Stan
         return MILLISECONDS.toDays(millis);
     }
 
+    @Description("Generate a date based on the year, month and day passed in")
+    @ScalarFunction("date")
+    @SqlType(StandardTypes.DATE)
+    public static long toDate(@SqlType(StandardTypes.BIGINT) long year, @SqlType(StandardTypes.BIGINT) long month, @SqlType(StandardTypes.BIGINT) long day)
+    {
+        try {
+            return MILLISECONDS.toDays(UTC_CHRONOLOGY.getDateTimeMillis(toIntExact(year), toIntExact(month), toIntExact(day), 0));
```

Comment:
What happens with a DATE in the Gregorian-Julian cutoff? Also, negative dates (before 0 A.D.)?

Add tests for those.

Reply:
@hashhar Done.